Feature/place management

park0503 opened the pull request:
### 🔨 작업 내용
- PlaceService에서 Dto를 사용하도록 구조 변경 및 그에 따른 Test 수정
- KakaoMap Api로부터 받아올 data를 위해 place entity에 kakaoMapId 칼럼 추가
- Schedule 생성시 Plcae에 대한 처리 로직을 find or create로 변경
### 🚧 논의 사항
- x

park0503 committed:
refactor: schedule 생성 시 place에 대한 처리를 find or create 로 변경
- fix: place 수정으로 발생한 test bug fix

park0503 committed:
refactor: placeService 내부 로직 dto를 사용하도록 구조 변경

park0503 committed:
refactor: PlaceResponseDto에 kakaoMapId 칼럼 추가

park0503 committed:
fix: pull and fix conflict

park0503 committed:
feat: shceulde update test 구현

park0503 committed:
refactor: schedule 업데이트 시 place에 대한 처리를 find or create로 변경
- fix: pull and fix conflict

park0503 committed:
Merge pull request #32 from Ajou-Travely/feature/schedule-update-test

feat: shceulde update test 구현

park0503 commented:
Schedule Update 시에도 Place에 대한 처리를 Create or Find 로직으로 처리하게 수정했고
Schedule Update 에 대한 Test 코드 또한 구현했습니다.

Sophoca commented on src/main/java/com/ajou/travely/service/ScheduleService.java:
```diff
@@ -73,10 +83,23 @@ public ScheduleResponseDto updateSchedule(ScheduleUpdateRequestDto scheduleUpdat
+        schedule
+                .getTravel()
+                .getUserTravels()
+                .forEach(ut -> currentUsers.put(ut.getUser().getId(), ut.getUser()));
```
ut -> userTravel로 수정 부탁드립니다.

park0503 replied:
확인했습니다.

Sophoca commented on src/main/java/com/ajou/travely/service/ScheduleService.java:
```diff
@@ -73,10 +83,23 @@ public ScheduleResponseDto updateSchedule(ScheduleUpdateRequestDto scheduleUpdat
+                            .addressName(placeDto.getAddressName())
+                            .x(placeDto.getX())
+                            .y(placeDto.getY())
+                            .build()));
```
위에서 중복되는 코드가 있는 것으로 확인됩니다.
`getOrCreatePlace()` 같은 이름으로 메소드 분리가 가능할 것 같습니다.

ErranderLee commented on src/main/java/com/ajou/travely/service/ScheduleService.java:
```diff
@@ -73,7 +82,10 @@ public ScheduleResponseDto updateSchedule(ScheduleUpdateRequestDto scheduleUpdat
+        schedule
+                .getTravel()
+                .getUserTravels()
+                .forEach(ut -> currentUsers.put(ut.getUser().getId(), ut.getUser()));
```
반복문 내의 반복 대상이 되는 변수명을 좀 더 명확하게 하시면 좋을 것 같아요 ㅎㅎ

park0503 replied:
확인했습니다

ErranderLee reviewed (commented):
고생하셨습니다.

Sophoca reviewed (commented):
고생하셨습니다.

park0503 committed:
refactor: 코드 리뷰 피드백 반영
- refactor: schedule create url 및 파라미터 변경
- refactor: schedule update url 및 파라미터 변경

park0503 committed:
refactor: 코드 리뷰 피드백 반영
